libkbfs/coalescing_context.go

```diff
@@ -0,0 +1,110 @@
+package libkbfs
+
+import (
+	"golang.org/x/net/context"
+	"reflect"
+)
+
+// CoalescingContext allows many contexts to be treated as one.  It waits on
+// all its contexts' Context.Done() channels, and when all of them have
+// returned, this CoalescingContext is canceled. At any point, a context can be
+// added to the list, and will subsequently also be part of the wait condition.
+// TODO: add timeout channel in case there is a goroutine leak
+type CoalescingContext struct {
+	context.Context
+	closeCh  chan struct{}
```

Comment:
Right but that's done in the constructor, and can be done via a local var rather than saved that way in the struct.

Super minor point though, not really worth these bits.